src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHelpers.Unix.cs

```diff
@@ -47,57 +47,41 @@ public int Compare (string? x, string? y)
 
         private static UnixFileMode UMask => s_umask.Value;
 
-        /*
-            Tar files are usually ordered: parent directories come before their child entries.
-
-            They may be unordered. In that case we need to create parent directories before
-            we know the proper permissions for these directories.
-
-            We create these directories with restrictive permissions. If we encounter an entry for
-            the directory later, we store the mode to apply it later.
-
-            If the archive doesn't have an entry for the parent directory, we use the default mask.
-
-            The pending modes to be applied are tracked through a reverse-sorted dictionary.
-            The reverse order is needed to apply permissions to children before their parent.
-            Otherwise we may apply a restrictive mask to the parent, that prevents us from
-            changing a child.
-        */
-
+        // Use a reverse-sorted dictionary to apply permission to children before their parents.
+        // Otherwise we may apply a restrictive mask to the parent, that prevents us from changing a child.
         internal static SortedDictionary<string, UnixFileMode>? CreatePendingModesDictionary()
             => new SortedDictionary<string, UnixFileMode>(s_reverseStringComparer);
 
-        internal static void CreateDirectory(string fullPath, UnixFileMode? mode, bool overwriteMetadata, SortedDictionary<string, UnixFileMode>? pendingModes)
+        internal static void CreateDirectory(string fullPath, UnixFileMode? mode, SortedDictionary<string, UnixFileMode>? pendingModes)
```

Comment:
You need to also change this signature in TarHelpers.Windows.cs.

Reply:
Thanks. Can you trigger the test run again please?